fastar/core.py

```diff
@@ -84,10 +84,9 @@ def toposort(haxpr):
   while childless_boxes:
     for e_sorted, e in reversed(zip(sorted_updates, haxpr.eqns)):
       o, = e.outvars
-      if o in childless_boxes:
-        obox = childless_boxes[o].pop()
-        if not childless_boxes[o]:
-          childless_boxes.pop(o)
+      oboxes = childless_boxes.get(o)
+      if oboxes:
+        obox = oboxes.pop()
```

Comment:
Need to be careful with changing this logic. I think the way it is in your proposed change could result in all of the lists  having a `None` at the end - once each list in childless boxes is empty there can still be a loop through the haxpr where every list would have a `None` added to it.

I think the right way to fix the bug you're hitting is to prevent empty lists being added to the dict [here](https://github.com/j-towns/fastar/blob/c97772f4d25154daf41530c508a1d141a7994183/fastar/core.py#L82-L83), i.e. you could first compute the `static_box_finder`, then assign it the `childless_boxes` dict only if the list is non-empty. Does that make sense?